components/post_view/channel_intro_message/boards_button.tsx

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+
+import {FormattedMessage} from 'react-intl';
+
+import {PluginComponent} from 'types/store/plugins';
+import * as Utils from 'utils/utils';
+
+type Props = {
+    boardComponent?: PluginComponent;
+}
+
+const BoardsButton = ({boardComponent}: Props) => {
+    if (!boardComponent) {
+        return null;
+    }
+
+    return (
+        <button
+            className={'intro-links color--link channelIntroButton style--none'}
+            onClick={() => boardComponent.action?.()}
+            aria-label={Utils.localizeMessage('intro_messages.createBoard', 'Create a board')}
+        >
+            {boardComponent.icon}
+            <FormattedMessage
+                id='intro_messages.createBoard'
+                defaultMessage='Create a board'
+            />
```

Comment:
```suggestion
            {formatMessage({id: 'intro_messages.createBoard', defaultMessage: 'Create a board'})}
```

Reply:
well ... both are valid ... you can still use the declarative approach ... but since we know that this is a pure string we can just use the imperative one here as well ... it basically boils down to personal preference

Reply:
My rule of thumb is that `FormattedMessage` is the best to use followed by `useIntl`/`formatMessage` when you need a string. `Utils.localizeMessage` should only be used when neither of those are an option